### DISCUSSION TITLE
2024-07-16 화요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#27] [기록보드]
2024-07-16 회고입니다

## 🍀 공통 일과표

| 시간 | 진행내용 | 기록 |
|:-:|:-|:-:|
| 09:00 ~ 10:00 | 알고리즘 특강 | 
| 10:10 ~ 11:10 | 최종 프로젝트 발제 | <img src="https://github.com/user-attachments/assets/c88804ae-1f94-42b0-8b5d-31c1c10b9590" width="150" height="150"/> |
| 11:10 ~ 11:30 | 반응형 웹 특강 | <img src="https://github.com/user-attachments/assets/bacd9970-4d18-4925-b99b-d1f087cf52a8" width="150" height="150"/> |
| 11:30 ~ 13:00 | 오전 기획 회의(디자이너 동반) | <img src="https://github.com/user-attachments/assets/19703d7d-082f-4f3c-800e-82c13ed630b5" width="150" height="150"/> |
| 14:00 ~ 18:00 | 오후 기획 회의(디자이너 동반) : 프로젝트명, 유저플로우 | <img src="https://github.com/user-attachments/assets/122831b9-93e4-41b3-ba59-63d9ef273340" width="150" height="150"/> <img src="https://github.com/user-attachments/assets/b9a248d5-b7ac-4a8f-886a-625fe0fe709e" width="150" height="150"/>|
| 19:00 ~ 21:00 | 심야 기획 회의(개발자) : User Flow Chart 작성, 도전 과제 분배 | <img src="https://github.com/user-attachments/assets/4ecdb488-cc26-4320-973f-2be074f95139" width="150" height="150"/> <img src="https://github.com/user-attachments/assets/27833b27-bdb2-4d0d-a77a-01f455c0dd96" width="150" height="150"/>|

**Comment by Captain-Kim** [comment]
작업자 : @eunohhh @Captain-Kim 

**PWA 셋업 및 설치 유도 버튼 제작**

- WEB에서는 동작 잘 하는 것으로 확인
- Mobile에서는 동작하지 않는 것으로 확인

<table>
    <tr>
        <td><a href="https://github.com/user-attachments/assets/fbe2761d-35ab-48e5-9f18-a097e11c825a"><img src="https://github.com/user-attachments/assets/fbe2761d-35ab-48e5-9f18-a097e11c825a" alt="Image 1" width="300" height="300"></a></td>
        <td><a href="https://github.com/user-attachments/assets/98e0d9bd-11f5-479f-b9ad-551f6f6d029"><img src="https://github.com/user-attachments/assets/98e0d9bd-11f5-479f-b9ad-551f6f6d0297" alt="Image 1" width="300" height="300"></a></td>
    </tr>
</table>

**Comment by slsl2** [comment]
**기획 회의 (다같이)**

- 아이디어 디벨롭, 레퍼런스 서치, 피그마에 정리
- 프로젝트 이름 브레인스토밍 및 확정
- 챌린지 기능 회의, 역할 분배
- IA, 유저 플로우 작성 시작

![image 3](https://github.com/user-attachments/assets/40605366-e1b1-4a67-91b2-dcc209f0b232)

**Comment by eunohhh** [comment]
pwa 모바일 테스트 하다가 알게된 부분 공유합니다

- 현재 'beforeinstallprompt' 이벤트는 실험적 기능으로 모바일에서는 안드로이드 크롬에서만 작동합니다.
- 따라서 web share api[참고링크](https://developer.mozilla.org/en-US/docs/Web/API/Web_Share_API)를 사용하여, 사용자액션을 유도하는 방식으로 시도해야할 것 같습니다.
- web share api 를 로컬에서 테스트하려면, secure 모드 즉 로컬에서도 https 접속이 가능해야 합니다.
- 이를 위한 코드는 chore/pwa-test 브랜치에서 확인하실 수 있습니다.
- 해당 브랜치에서는 yarn init-https 명령어를 입력하시면 로컬에서 https 모드 세팅이 완료되고(주의! 맥만 가능, 윈도우 사용자는 알려주세요)
- yarn secure 로 dev server 실행하실 수 있습니다.
